### PR TITLE
enable multiple cloud profile with the same subscription Id

### PR DESCRIPTION
**Pull request by juniwang**
[FIXED JENKINS-43704]

- update the name for AzureVMCloud to use the combination of resourceGroupName and subscriptionId. Since resourceGroupName is unique in every subscripton, it should cover the scenario in the issue. If user try to add a new cloud profile with the same sub Id and the same resourceGroupName, the latter will be useless, Jenkins will always make use the first one. 
- Similarly for agetn template name, it could be the same among agent templates within different clouds. By using the combination of template name and cloud name, we can make sure to pick up the right template during verification process.  Again, if templates with same name exist in one single cloud, only the first templates got verified.

### CODE REVIEW

File: src/main/java/com/microsoft/azure/vmagent/AzureVMAgent.java [hunk 1]
```diff
@@ -259,9 +259,6 @@ public Localizable getCleanUpReason() {
         return cleanUpReason;
     }
 
-    /**
```
Comment (clguimanMSFT) [inline src/main/java/com/microsoft/azure/vmagent/AzureVMAgent.java]: nitpick: please just commit file that actually changed and not whitespace.

**Comment by clguimanMSFT** [comment]
LGTM. Have you verified this won't brake when the new version is installed while Jenkins already has agents deployed?

**Comment by juniwang** [comment]
@clguimanMSFT Yes, it's verified that an newer version of vm agent plugin won't broken the existing configurations.

File: src/main/java/com/microsoft/azure/vmagent/util/AzureUtil.java [hunk 2]
```diff
@@ -242,16 +242,17 @@ public static boolean isDeploymentAlreadyOccupied(String errorMessage) {
...
     /**
      * Retrieves the name of the cloud for registering with Jenkins
      * @param subscriptionId Subscription id
+     * @param resourceGroupName Resource group name
      * @return Name of the cloud
      */
-    public static String getCloudName(String subscriptionId) {
-        return Constants.AZURE_CLOUD_PREFIX + subscriptionId;
+    public static String getCloudName(final String subscriptionId, final String resourceGroupName) {
+        return Constants.AZURE_CLOUD_PREFIX + subscriptionId + resourceGroupName;
```
Comment (chenkennt) [inline src/main/java/com/microsoft/azure/vmagent/util/AzureUtil.java]: Add a separator between `subscriptionId` and `resourceGroupName`?

Comment (mmitche) [inline src/main/java/com/microsoft/azure/vmagent/util/AzureUtil.java]: Yeah, adding a separator would be best.

**Comment by clguimanMSFT** [comment]
please add the ticket number in your commit message (append '[FIXED JENKINS-43704]'), so the ticket gets closed automatically

**Pull request merged**